### PR TITLE
Updating 2x branch with the net9 changes currently present in features/net9 branch

### PR DESCRIPTION
**Pull request by kshyju**
Updating 2x branch with the net9 changes currently present in features/net9 branch. This brings the below 2 commits.

  - #2597
  - #2615 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj [hunk 1]
```diff
@@ -2,15 +2,14 @@
 
   <PropertyGroup>
     <OutputType>Library</OutputType>
-    <TargetFrameworks>net5.0;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net5.0;net8.0;net9.0;netstandard2.0</TargetFrameworks>
```
Comment (kshyju) [inline src/DotNetWorker.Core/DotNetWorker.Core.csproj:5]: I did not remove net5.0 as part of this PR. We have a GH work item to track that change https://github.com/Azure/azure-functions-dotnet-worker/issues/1095

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj [hunk 2]
```diff
@@ -22,6 +21,16 @@
...
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Collections.Immutable" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="9.0.0-preview.*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
```
Comment (kshyju) [inline src/DotNetWorker.Core/DotNetWorker.Core.csproj:34]: As part of #1095 we should discuss we need to have 8x,7x and 6x version of these dependencies updated  similar to what we are doing here for net9.

**Comment by fabiocav** [comment]
Are the commits here in the right state for merging?

**Comment by kshyju** [comment]
> Are the commits here in the right state for merging?

YES

File: samples/Net9FunctionApp/Net9FunctionApp.csproj [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+  <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.2.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore" Version="1.3.2" />
+    <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.22.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.ApplicationInsights" Version="1.3.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.17.0" />
```
Comment (mattchenderson) [inline samples/Net9FunctionApp/Net9FunctionApp.csproj:15]: This should reference the version defined in this PR 
```suggestion
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.18.0-preview1" />
```

File: src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj [hunk 4]
```diff
@@ -27,6 +26,14 @@
...
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
```
Comment (mattchenderson) [inline src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj:36]: If we're adding the .NET 8 target with this PR, should we also do the same thing for .NET 8 here?

File: src/DotNetWorker/DotNetWorker.csproj [hunk 5]
```diff
@@ -2,21 +2,28 @@
...
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.41.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
```
Comment (mattchenderson) [inline src/DotNetWorker/DotNetWorker.csproj:26]: If we're adding the .NET 8 target with this PR, should we also do the same thing for .NET 8 here?

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj [hunk 6]
```diff
@@ -22,6 +21,16 @@
...
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Collections.Immutable" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="9.0.0-preview.*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
```
Comment (mattchenderson) [inline src/DotNetWorker.Core/DotNetWorker.Core.csproj:33]: If we're adding the .NET 8 target with this PR, should we also do the same thing for .NET 8 here?

**Comment by kshyju** [comment]
> > Are the commits here in the right state for merging?
> 
> ~YES~ NO. LET ME CLEAN THIS UP

Updated the PR to have only those 2 commits cherry picked and resolved merge conflicts. This is in a clean state to merge.
FYI @mattchenderson

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj (hunk 6 above)
Comment (kshyju) [reply src/DotNetWorker.Core/DotNetWorker.Core.csproj:34]: So, I updated the PR to pick only those 2 commits. The changes you suggested can go as a separate commit.

**Review by fabiocav — APPROVED** [review]

File: samples/Net9FunctionApp/Net9FunctionApp.csproj (hunk 3 above)
Comment (fabiocav) [reply samples/Net9FunctionApp/Net9FunctionApp.csproj:15]: @kshyju will be following up with changes against 2.x

File: src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj (hunk 4 above)
Comment (fabiocav) [reply src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj:37]: @kshyju will be following up with changes against 2.x

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj (hunk 6 above)
Comment (fabiocav) [reply src/DotNetWorker.Core/DotNetWorker.Core.csproj:34]: @kshyju will be following up with changes against 2.x

File: src/DotNetWorker/DotNetWorker.csproj (hunk 5 above)
Comment (fabiocav) [reply src/DotNetWorker/DotNetWorker.csproj:27]: @kshyju will be following up with changes against 2.x

**Pull request merged**